libpod/container_internal.go

```diff
@@ -2738,3 +2741,365 @@ func (c *Container) update(resources *spec.LinuxResources, restartPolicy *string
 
 	return nil
 }
+
+func (c *Container) getCopyOfHealthCheckAndStartupHelathCheck() (*manifest.Schema2HealthConfig, *define.StartupHealthCheck) {
+	var healthCheck manifest.Schema2HealthConfig
+	if c.config.HealthCheckConfig != nil {
+		healthCheck = manifest.Schema2HealthConfig{
+			Test:          slices.Clone(c.config.HealthCheckConfig.Test),
+			Interval:      c.config.HealthCheckConfig.Interval,
+			Retries:       c.config.HealthCheckConfig.Retries,
+			StartInterval: c.config.HealthCheckConfig.StartInterval,
+			StartPeriod:   c.config.HealthCheckConfig.StartPeriod,
+			Timeout:       c.config.HealthCheckConfig.Timeout,
+		}
+	}
+
+	var startupHealthCheck define.StartupHealthCheck
+	if c.config.StartupHealthCheckConfig != nil {
+		startupHealthCheck = define.StartupHealthCheck{
+			Successes: c.config.StartupHealthCheckConfig.Successes,
+		}
+		startupHealthCheck.Test = slices.Clone(c.config.StartupHealthCheckConfig.Test)
+		startupHealthCheck.Interval = c.config.StartupHealthCheckConfig.Interval
+		startupHealthCheck.Retries = c.config.StartupHealthCheckConfig.Retries
+		startupHealthCheck.StartInterval = c.config.StartupHealthCheckConfig.StartInterval
+		startupHealthCheck.StartPeriod = c.config.StartupHealthCheckConfig.StartPeriod
+		startupHealthCheck.Timeout = c.config.StartupHealthCheckConfig.Timeout
+	}
+	return &healthCheck, &startupHealthCheck
+}
+
+func containsFlags(changedHealthCheckConfig *entities.UpdateHealthCheckConfig, expectedFlags []string) bool {
+	containsFlag := false
+	for _, v := range expectedFlags {
+		if _, contains := (*changedHealthCheckConfig)[v]; contains {
+			containsFlag = true
+		}
+	}
+	return containsFlag
+}
+
+func (c *Container) changeHealthCheckConfiguration(changedHealthCheckConfig *entities.UpdateHealthCheckConfig) (bool, error) {
+	_, containsHealthCheckCmd := (*changedHealthCheckConfig)["HealthCmd"]
+	containsFlags_ := containsFlags(changedHealthCheckConfig, []string{"HealthInterval", "HealthRetries", "HealthTimeout", "HealthStartPeriod"})
+	if c.config.HealthCheckConfig == nil && !containsHealthCheckCmd && containsFlags_ {
+		return false, errors.New("healthcheck command is not set")
+	}
+
+	cmd := ""
+	interval := define.DefaultHealthCheckInterval
+	retries := int(define.DefaultHealthCheckRetries)
+	timeout := define.DefaultHealthCheckTimeout
+	startPeriod := define.DefaultHealthCheckStartPeriod
+
+	if c.config.HealthCheckConfig != nil {
+		cmd = strings.Join(c.config.HealthCheckConfig.Test, " ")
+		interval = c.config.HealthCheckConfig.Interval.String()
+		retries = c.config.HealthCheckConfig.Retries
+		timeout = c.config.HealthCheckConfig.Timeout.String()
+		startPeriod = c.config.HealthCheckConfig.StartPeriod.String()
+	}
+
+	changed := false
+	changedTimer := false
+	noHealthCheck := false
+	for k, v := range *changedHealthCheckConfig {
+		switch k {
+		case "NoHealthCheck":
```

Comment:
all these strings seem to be used in many places so can you define them as constants somewhere, ie. libpod/define and then use the constants over the string duplication

Reply:
Thanks, I have an idea of how to simplify it overall.